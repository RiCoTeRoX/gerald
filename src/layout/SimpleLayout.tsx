import React, { FC, ReactNode } from 'react'
import { StyleSheet } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

type SimpleLayoutProps = {
  children: ReactNode
  bg?: string
}

const SimpleLayout: FC<SimpleLayoutProps> = ({ children, bg }) => {
  return (
    <SafeAreaView style={[styles.container, { backgroundColor: bg }]} edges={{ top: 'maximum', bottom: 'maximum' }}>
      {children}
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderRadius: 20,
    paddingHorizontal: 26,
  },
})

export default SimpleLayout
