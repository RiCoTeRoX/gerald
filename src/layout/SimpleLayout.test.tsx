import { render } from '@testing-library/react-native'
import React from 'react'
import { Text, View } from 'react-native'

import SimpleLayout from './SimpleLayout'

describe('SimpleLayout', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <SimpleLayout>
        <View>
          <Text>Test</Text>
        </View>
      </SimpleLayout>
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
