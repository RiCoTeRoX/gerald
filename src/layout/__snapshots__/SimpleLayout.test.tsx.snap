// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleLayout renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "maximum",
      "left": "off",
      "right": "off",
      "top": "maximum",
    }
  }
  style={
    [
      {
        "borderRadius": 20,
        "flex": 1,
        "paddingHorizontal": 26,
      },
      {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <View>
    <Text>
      Test
    </Text>
  </View>
</RNCSafeAreaView>
`;
